cmd/gokey/main.go

```diff
@@ -79,20 +83,43 @@ func genKey(seed []byte, w io.Writer) {
 	}
 }
 
-// TODO: parametrize size
-// generates raw 32 bytes
 func genRaw(seed []byte, w io.Writer) {
+	if isInvalidLength(length) {
+		log.Fatalln("Invalid Length")
+	}
+	if !isFlagPassed("l") {
+		length = 32
+	}
+
 	raw, err := gokey.GetRaw(pass, realm, seed, unsafe)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
-	_, err = io.CopyN(w, raw, 32)
+	_, err = io.CopyN(w, raw, int64(length))
 	if err != nil {
 		log.Fatalln(err)
 	}
 }
 
+func isFlagPassed(name string) bool {
+	found := false
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == name {
+			found = true
+		}
+	})
+	return found
+}
+
+func isInvalidLength(l int) bool {
```

Comment:
my preference is to drop this function and just do the validation inline in the `main` before calling the helper functions

Reply:
Resolved in 2cb0070